7_react/react1/week3/vite-project/src/App.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from "react";
+import "./App.css";
+
+import AddTodo from "./components/AddTodo";
+import TodoList from "./components/TodoList";
+import Timer from "./components/Timer";
+
+const App = () => {
+  const [todos, setTodos] = useState([]);
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setTodos(data);
```

Comment:
If this `fetch` was very slow, it could potentially overwrite todos created by the user! 😮 Can you think of a way to avoid this happening?

Reply:
@NVedsted  I did some changes,
 you can have a look?🥺